### PR TITLE
[ws-deployment] Introduce context wrapper and overrides

### PR DESCRIPTION
**Pull request by princerachit**
## Description

Introduced a wrapper struct `Context` to wrap all kinds of context. This will reduce argument size of various methods.
Introduced a new struct `Overrides` that can be used to override the default behaviour of deployment.
Updated methods which are affected by above change.

## Related Issue(s)

Fixes #

## How to test

use the flags `--dry-run` and you should not see any resources being created, however the relevant artifacts like terraform scripts will be created.
I have tested it here https://werft.gitpod-io-dev.com/job/ops-workspace-deploy-workspace-main.14/raw

To test `--overwrite` flag you can use werft sideload feature to overwrite the name of the clusters in the config.

## Release Notes

```release-note
None
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **princerachit** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[operations/workspace/deployment/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/workspace/deployment/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: operations/workspace/deployment/pkg/common/context.go [hunk 1]
```diff
@@ -25,9 +30,3 @@ type GitpodContext struct {
 	// configuration of gitpod
 	ValuesFiles []string
 }
-
-// ClusterContext contains the context to access the cluster
-type ClusterContext struct {
```
Comment (princerachit) [inline operations/workspace/deployment/pkg/common/context.go:30]: This was unused so got rid of it. Will introduce later if required.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: cf3e9d7e8737d29a4628d379852489346c86d90b</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6497#issuecomment-957826292" title="LGTM">csweichel</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[operations/workspace/deployment/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/workspace/deployment/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6497#issuecomment-957826516" title="Approved">csweichel</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6497#issuecomment-957826516" title="Approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[operations/workspace/deployment/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/operations/workspace/deployment/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**